Formula/haste-client.rb

```diff
@@ -26,20 +26,29 @@ class HasteClient < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "f58ba7733cf6b73c7ad7be755d4d31c88af9577a43f9cd4658d8732bdd8bad3b"
   end
 
-  depends_on "ruby" if MacOS.version <= :sierra
+  on_linux do
+    depends_on "ruby"
 
-  resource "faraday" do
-    url "https://rubygems.org/gems/faraday-0.17.4.gem"
-    sha256 "11677b5b261fbbfd4d959f702078d81c0bb66006c00ab2f329f32784778e4d9c"
+    resource "json" do
+      url "https://rubygems.org/gems/json-2.5.1.gem"
+      sha256 "918d8c41dacb7cfdbe0c7bbd6014a5372f0cf1c454ca150e9f4010fe80cc3153"
+    end
   end
 
-  if MacOS.version <= :sierra
+  on_sierra :or_older do
+    depends_on "ruby"
+
     resource "json" do
       url "https://rubygems.org/gems/json-2.5.1.gem"
       sha256 "918d8c41dacb7cfdbe0c7bbd6014a5372f0cf1c454ca150e9f4010fe80cc3153"
     end
   end
```

Comment:
Strong 👎 on having to declare the same resource twice in different blocks.

I don't have ideas for how to do it better yet, but we should discuss it.